### DISCUSSION TITLE
Next release date?

### DISCUSSION THREAD

**Discussion by depler** [transmission/transmission#2287] [General]
When is it going to release next version of transmission? Is there any roadmap?

**Comment by ckerr** [comment]
Definitely in 2022 :laughing: 

There are a couple of more memory improvements that I want to land before we start the release treadmill. the `tr_torrent_metainfo` refactor that happened over Dec / Jan was a major prerequisite for this, so things are moving forward but I don't have a date in mind yet. My _best guess_ is sometime in Q1 but I can't promise anything. (*edit:* I got the estimate wrong when asked this late last year; I estimated January :laughing:)

Also I think we should consider making the next release `4.0.0-beta.1` or something along those lines -- there's been a very significant amount of rewriting & refactoring, so I think the  major version bump _and_ a beta cycle are both warranted. There's also been some discussion / user requests about moving to semver, and we would need a major bump in order to make that move.

**Comment by bluskye** [comment]
This would be so amazing.  Been using transmission in docker for years, would love a new release.  Its so stable, and sips resources compared to my hellish experience now with anything using libtorrent.

**Comment by depler** [comment]
The release button

![baebd6783416471e8999a1adfd77219e](https://user-images.githubusercontent.com/13541699/150644553-d3b3987f-d126-4afe-a3d1-dd4f246cb547.jpg)

**Reply by nautiluschan** [reply]
Just do  it!

**Reply by depler** [reply]
![IMG_20220724_112428](https://user-images.githubusercontent.com/13541699/180638799-95151a0f-46fb-4e3e-ae45-db0e3d6b0643.jpg)

**Reply by nevack** [reply]
Why don't you just download Nightly build? 4.0 would not have much difference from Nightly if it's released right now.

**Reply by woojh3690** [reply]
I think because of update. Manual install not support update. But package manager(apt ,yum) install support update.

**Reply by ckerr** [reply]
@depler There are like 5 commits landing each day! 

I agree the release needs to come sooner rather than later, but it's not like we're all asleep over here :smile_cat:

**Reply by depler** [reply]
@ckerr I just a little bit concerned that 2022 is ending, 5 months left 😊

**Reply by ckerr** [reply]
I laughed, but seriously, little "bump" comments like this make me want to unsub from this thread since the bumps are not productive. The development is happening at the rate that it's happening, so bumping is not productive. If you want it to happen faster, test the nightly builds and report any bugs that you find :heart:

**Comment by FluxState** [comment]
There is a Dev/Edge version of Dockerfile and build of Transmission. cc @bluskye 

https://github.com/FluxState/docker-transmission-vpn/blob/edge/Dockerfile
https://hub.docker.com/r/fluxstate/transmission-vpn-edge

**Reply by bluskye** [reply]
Thanks, I use the linuxserver.io container without VPN.  Ill sit and wait for a new stable transmission to make it downstream.  No rush for me, it just works!

**Comment by glefand** [comment]
Hi @ckerr , well Q1 is gone, but do you think we could have a beta in June ?
I would be more than happy to test it (mac os M1) and remove one more app using Rosetta :-)

**Reply by GaryElshaw** [reply]
I think of the nightlies as an extended beta :-) 
https://build.transmissionbt.com/job/trunk-mac/

**Reply by glefand** [reply]
I've switched... so far one system freeze and one comment : why colors of UL and DL stats on the icon have changed ? it's less readable I think

**Comment by frankyw** [comment]
It's now been over 2 years since a release, with lots of fixes since then. The release cycles need to be much shorter imo, while still allowing for stable LTS-style releases in between minor releases.

I love Transmission, but do I really need to switch to using the dev branch to get a fix made 6 months ago that still causes issues?

**Comment by ckerr** [comment]
> I love Transmission, but do I really need to switch to using the dev branch to get a fix made 6 months ago that still causes issues?

For now, yes. This switch from C to C++ has been a very positive development, but it's not without short-term pain: it's taken a lot of work to get the code production-ready, and that work generally doesn't backport easily to the 3.0.0 C codebase for bugfix releases.

IMO the project is getting close to being ready for a 4.0.0-beta.1 and I expect things to flow more steadily after that first beta is out the door.

There are still some medium-sized refactors that are probably more appropriate to land _before_ the beta cycle starts. There are some known bugs that would be nice-to-have before the betas, but probably aren't blockers for a beta.

**Comment by depler** [comment]
@ckerr, any updates about release date? Is current code of `master` branch stable enough for everyday using or not?

**Reply by ckerr** [reply]
I'm using `main` every day.

I've been holding off on cutting a beta because there's still stuff landing every day, e.g. there's been a lot of work to improve the look of the app on macOS. But in terms of stability,  the nightlies are better than 3.00 in my experience

**Reply by frankyw** [reply]
Sounds like it is ready for a beta binary to be released, for those of us who aren't able to make our own build?

**Reply by ckerr** [reply]
There are nightly builds available on transmisssionbt.com

**Reply by depler** [reply]
@frankyw what OS you need binaries for?

**Reply by frankyw** [reply]
@depler was looking for Ubuntu binaries for linuxserver docker container, but I'm either missing something, or the trunk-linux nightly tar file doesn't include binaries

**Reply by rezad1393** [reply]
> There are nightly builds available on transmisssionbt.com

I dont see any linux binary nightly builds there.

**Reply by fredo-47** [reply]
It's quite simple on Linux to build your own binaries from the nightly sources.
The exact commands are there on the main github/transmission page.

If you want to build only selected components, look [here.](https://github.com/transmission/transmission/discussions/2857)

And if you use the final command with option "DESTDIR=/path", then it would be installed below "/path" (i.e. "/path/usr/local/..." instead of "/usr/local/..." which may not even need sudo).
Example: 
`make DESTDIR=/tmp install`
This way, your existing files (from your distro) would remain as they are, i.e. not get overwritten. More convenient for testing nightlies :-)

**Comment by frozenpandaman** [comment]
I'd like to repeat one point from an earlier discussion here, since it doesn't seem to have been mentioned: Some sites only have certain torrent clients whitelisted, which tend to not be nightly versions of software. As such, I'm very much looking forward to the next official, stable release! :)

**Reply by valynor** [reply]
Yes, what frozenpandaman said. Alpha/beta/nightly are a huge no-no on a lot of sites. 

Please consider doing smaller updates but more frequently.

**Reply by Pentaphon** [reply]
> Alpha/beta/nightly are a huge no-no on a lot of sites.

I think the beta is just for this huge new 4.0.0 release since so much changed and they want to make sure this release is not riddled with issues like 3.0 was. After that its' going to be SemVer so you'll see 4.0.1 and so on.

**Comment by Pentaphon** [comment]


**Reply by depler** [reply]
Sorry, I'm using Windows 10 for now

**Comment by Pentaphon** [comment]
So just to be clear: what issues need to be resolved before 4.0.0 beta 1 comes out? 

The milestone seems to be complete except for Semver which is already done but there's more commits almost every day. 

Shouldn't this milestone list any remaining issues in order to attract contributors?

https://github.com/transmission/transmission/milestone/6

**Comment by ilf** [comment]
Thank you for [4.0.0-beta.1](https://github.com/transmission/transmission/releases/tag/4.0.0-beta.1)! <3

Do you have a roadmap or an ETA for a "final" 4.0.0?

Debian 12 "Bookworm" will be [frozen with the beginning of 2023](https://release.debian.org/testing/freeze_policy.html), and it would be super-awesome to have transmission 4.0 in it!

Thanks for all your work!

**Reply by ckerr** [reply]
> Thank you for [4.0.0-beta.1](https://github.com/transmission/transmission/releases/tag/4.0.0-beta.1)! <3

Thanks!
 
> Do you have a roadmap or an ETA for a "final" 4.0.0?

No. It depends on what kind of feedback / bug reports we get from the betas.
 
> Debian 12 "Bookworm" will be [frozen with the beginning of 2023](https://release.debian.org/testing/freeze_policy.html), and it would be super-awesome to have transmission 4.0 in it!

I agree, that would be great. But let's get some beta tester feedback before anything else. :test_tube:

**Comment by HanaO00** [comment]
In the release note for the daemon version, it says "better dark mode support", will dark mode be implemented or will only the color of the white page be improved for dark browsers?

Will it be possible to create torrents from the webui?

Thanks

**Comment by nafanz** [comment]
How about a roadmap?

To understand approximate dates, for example "release planned this year" is already better than no dates at all. It would also be good to see a list of new functionality, bug fixes that will be included. And current release blocking issues.

Debian 13 Trixie was frozen last month, which means its release and the derivative distribution Raspberry Pi OS will be released soon. I hope to see the future version of Transmission there.

**Reply by David-Else** [reply]
> Debian 13 Trixie was frozen last month, which means its release and the derivative distribution Raspberry Pi OS will be released soon. I hope to see the future version of Transmission there.

And it is [currently shipping the beta2](https://packages.debian.org/source/testing/transmission) with this super annoying bug https://github.com/transmission/transmission/issues/7572

**Reply by David-Else** [reply]
They fixed it with 4.1.0~beta2+dfsg-3! https://metadata.ftp-master.debian.org/changelogs//main/t/transmission/transmission_4.1.0~beta2+dfsg-3_changelog
